src/Symfony/Component/Translation/Translator.php

```diff
@@ -372,6 +372,39 @@ private function doLoadCatalogue($locale)
                 $this->catalogues[$locale]->addCatalogue($this->loaders[$resource[0]]->load($resource[1], $locale, $resource[2]));
             }
         }
+
+        $this->resolveCatalogueMessages($this->catalogues[$locale]);
+    }
+
+    private function resolveCatalogueMessages($catalogue)
```

Comment:
Could be interesting to check the impact of this method in terms of performance as well. It makes several `preg_match_all()` function calls inside loops...